lib/toopher-web/OAuth.php

```diff
@@ -481,16 +481,18 @@ public function build_signature($signature_method, $consumer, $token) {
   private static function generate_timestamp() {
     return time();
   }
-
+    
   /**
-   * util function: current nonce
+   * util function: generate secure nonce
    */
-  private static function generate_nonce() {
-    $mt = microtime();
-    $rand = mt_rand();
-
-    return md5($mt . $rand); // md5s look nicer than numbers
+  private static function generate_nonce($size = 16) {
+    if (!function_exists('openssl_random_pseudo_bytes')) {
+      throw new \InvalidArgumentException('OpenSSL not supported.');
```

Comment:
Ok, but it's a bit more than a missing dependency - it means that their actual PHP was built without OpenSSL, so they need a whole new interpreter.  Maybe something like "This PHP was built without OpenSSL support.  Please install a PHP built with OpenSSL to use the Toopher-Wordpress plugin"